pkg/metrics_pusher/pusher.go

```diff
@@ -58,7 +61,17 @@ func (p *MetricsPusher) pushMetrics() {
 			return
 		}
 	} else {
-		resp, err := http.Get(p.SourceURL)
+		caCert, err := ioutil.ReadFile(p.Clientca)
```

Comment:
Please check for errors here.

Reply:
Error from binddata.go. Have updated it.

Reply:
That is not what I meant.  The code needs to check for errors on this call.

Reply:
Tested on local as well as on POD by copying the Binary to pod.

/tmp/metrics-pusher --destination-path=/api/v1/namespaces/openshift-roks-metrics/services/push-gateway:http/proxy/metrics/job/cluster-version-operator  --frequency=30s --source-url=https://localhost:9099/metrics  --kubeconfig=/etc/openshift/kubeconfig/kubeconfig **--client-ca-file=/etc/kubernetes/config/initial-ca.crt**

{"level":"info","ts":1627614306.1596313,"logger":"metrics-pusher","msg":"Polling for metrics","from":"https://localhost:9099/metrics","to":"/api/v1/namespaces/openshift-roks-metrics/services/push-gateway:http/proxy/metrics/job/cluster-version-operator"}
{"level":"info","ts":1627614306.1747801,"logger":"metrics-pusher","msg":"Pushing metrics"}


Error code when CA is not provided:

/tmp/metrics-pusher --destination-path=/api/v1/namespaces/openshift-roks-metrics/services/push-gateway:http/proxy/metrics/job/cluster-version-operator  --frequency=30s --source-url=https://localhost:9099/metrics  --kubeconfig=/etc/openshift/kubeconfig/kubeconfig

{"level":"error","ts":1627614283.3214455,"logger":**"metrics-pusher","msg":"Unable to read CA cert for fetching metrics: Please provide valid CA cert or path**","error":"open : no such file or direct